httprequest_lego_provider/dns.py

```diff
@@ -14,21 +14,29 @@
 from .settings import GIT_REPO_URL
 
 FILENAME_TEMPLATE = "{domain}.domain"
-SPLIT_GIT_REPO_URL = GIT_REPO_URL.split("@")
-REPOSITORY_BASE_URL = "@".join(SPLIT_GIT_REPO_URL[:2])
-REPOSITORY_USER = (
-    SPLIT_GIT_REPO_URL[0].split("//")[1]
-    if SPLIT_GIT_REPO_URL and len(SPLIT_GIT_REPO_URL[0].split("//")) > 2
-    else None
-)
-REPOSITORY_BRANCH = SPLIT_GIT_REPO_URL[2] if len(SPLIT_GIT_REPO_URL) > 2 else None
 RECORD_CONTENT = "{record} 600 IN TXT \042{value}\042\n"
 
 
 class DnsSourceUpdateError(Exception):
     """Exception for DNS update errors."""
 
 
+def _parse_repository_url(repository_url: str) -> Tuple[str, str, str | None]:
+    """Get the parsed connection details from the repository connection string.
+
+    Args:
+        repository_url: the repository's connection string.
+
+    Returns:
+        the repository user.
```

Comment:
This doesn't quite seem correct. We're returning a tuple of user, base_url and branch.